src/main/java/TaskToEventManual.java

```diff
@@ -13,12 +13,12 @@
      * @return an Event scheduled at a user suggested time
      */
     @Override
-    public Event createEventFromTask(Task task, Calendar calendar) {
+    public Event createEventFromTask(Task task, Calendar calendar, EventScheduler eventScheduler) {
         LocalDateTime userSuggestedTime = getUserSuggestedTime();
-        boolean validTime = calendar.checkAvailability(userSuggestedTime);
+        boolean validTime = eventScheduler.checkAvailability(userSuggestedTime, calendar);
         while (!validTime) {
             userSuggestedTime = getUserSuggestedTime();
-            validTime = calendar.checkAvailability(userSuggestedTime);
+            validTime = eventScheduler.checkAvailability(userSuggestedTime, calendar);
         }
         return new Event(task, userSuggestedTime, userSuggestedTime.toLocalTime().plus(task.getTimeNeeded()));
```

Comment:
Wait where is the user inputting tasks? Wouldn't the check if the task is valid be wherever that is?

Reply:
oh wait I just reread this, I understand what you're saying now. A prompt will need to be given to the user if the task doesn't have a time right? Like "input time needed:" type of thing.

Reply:
I added a check and a todo as I the prompt needs to be added into the command line interface

Reply:
Yes! Since TimeNeeded may not exist (or =0), in which case `userSuggestedTime` will be equal to `userSuggestedTime.toLocalTime().plus(task.getTimeNeeded()`.